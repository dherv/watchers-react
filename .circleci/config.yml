version: 2
jobs:
  build:
    working_directory: ~/tmp
    branches:
      only:
        - develop
        - feature/*
        - master
    docker:
      - image: circleci/node
    steps:
      - checkout
      - run: yarn install
      - run: yarn test
