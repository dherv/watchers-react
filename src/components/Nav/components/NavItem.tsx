import React, { PureComponent } from "react";
import styles from "./NavItem.module.css";

interface IProps {
  item: string;
  onClick(item: string): void;
}
export default class NavItem extends PureComponent<IProps> {
  constructor(props: IProps) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    this.props.onClick(this.props.item);
  }
  render() {
    return (
      <li className={styles.item}>
        <a className={styles.link} onClick={this.handleClick}>
          {this.props.item}
        </a>
      </li>
    );
  }
}
