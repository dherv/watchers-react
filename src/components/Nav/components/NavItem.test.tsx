import { renderer, shallow, React } from "../../../test.config";
import NavItem from "./NavItem";

describe("NavItem", () => {
  const props = {
    item: "item1",
    onClick: jest.fn()
  };
  const wrapper = shallow(<NavItem {...props} />);
  describe("component", () => {
    it("should display one li and one anchor", () => {
      expect.assertions(2);
      expect(wrapper.find("li").length).toBe(1);
      expect(wrapper.find("a").length).toBe(1);
    });

    it("should display the text from the props.item", () => {
      expect.assertions(1);
      expect(wrapper.find("a").text()).toBe("item1");
    });

    it("should call handleClick when anchor link is clicked", () => {
      expect.assertions(3);
      const handleClick = jest.spyOn(NavItem.prototype, "handleClick");
      const wrapper = shallow(<NavItem {...props} />);
      const link = wrapper.find("a");
      link.simulate("click");
      expect(handleClick).toHaveBeenCalledTimes(1);
      expect(props.onClick).toHaveBeenCalledTimes(1);
      expect(props.onClick).toHaveBeenCalledWith("item1");
    });
  });
  describe("snapshot", () => {
    test("should match", () => {
      const tree = renderer.create(<NavItem {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
