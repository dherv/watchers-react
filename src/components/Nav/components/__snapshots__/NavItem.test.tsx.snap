// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem snapshot should match 1`] = `
<li>
  <a
    className="link"
    onClick={[Function]}
  >
    item1
  </a>
</li>
`;
